File: src/main/java/com/gary/GalacticTrading/converter/IntergalacticUnitsToRomanStringConverter.java
package com.gary.GalacticTrading.converter;

import org.springframework.stereotype.Service;

import java.util.Map;

/**
 * This class is responsible for converting intergalactic units to roman string.
 */
@Service
public class IntergalacticUnitsToRomanStringConverter {
    private Map<String, String> interGalacticUnits;

    public IntergalacticUnitsToRomanStringConverter(Map<String, String> interGalacticUnits) {
        this.interGalacticUnits = interGalacticUnits;
    }

    /**
     * This method is used to convert intergalactic units to roman string.
     * @param interGalacticUnits
     * @return
     */
    public String convertIntergalacticUnitsToRomanString(String interGalacticUnits) {
        String[] interGalacticUnitArray = interGalacticUnits.split(" ");
        StringBuilder romanString = new StringBuilder();
        for (String interGalacticUnit : interGalacticUnitArray) {
            romanString.append(this.interGalacticUnits.get(interGalacticUnit));
        }
        return romanString.toString();
    }
}

File: src/main/kotlin/com/gary/GalacticTrading/converter/IntergalacticUnitsToRomanStringConverter.kt
package com.gary.GalacticTrading.converter

import mu.KotlinLogging
import org.springframework.stereotype.Service
import java.util.*

/**
 * This class is responsible for converting intergalactic units to roman string.
 */
@Service
class IntergalacticUnitsToRomanStringConverter() {
    val interGalacticUnits: MutableMap<String, String> = HashMap()
    private val log = KotlinLogging.logger {}

    fun setInterGalacticUnits(interGalacticUnits: MutableMap<String, String>) {
        this.interGalacticUnits.putAll(interGalacticUnits)
    }
    /**
     * This method is used to convert intergalactic units to roman string.
     * @param interGalacticUnits intergalactic units string to be converted to roman string
     * @return roman string
     */
    fun convertIntergalacticUnitsToRomanString(interGalacticUnits: String): String {
        val interGalacticUnitArray =
            interGalacticUnits.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val romanString = StringBuilder()
        for (interGalacticUnit in interGalacticUnitArray) {
            romanString.append(this.interGalacticUnits[interGalacticUnit.lowercase(Locale.getDefault())])
        }
        return romanString.toString()
    }
}
